### PR TITLE
Adds a --file flag to :jseval

### PR DESCRIPTION
**Pull request by pkill-nine**
#2349

### CODE REVIEW

File: qutebrowser/browser/commands.py [hunk 1]
```diff
@@ -1960,6 +1960,13 @@ def jseval_cb(out):
                         out = out[:5000] + ' [...trimmed...]'
                     message.info(out)
 
+        if file:
+            try:
+                with open(js_code) as f:
+                    js_code = f.read()
+            except FileNotFoundError:
```
Comment (The-Compiler) [inline qutebrowser/browser/commands.py]: Please handle `OSError` here instead to handle other issues while reading the file as well, and add the exception message into the error message (do `as e:` here and I'd say just do `message.error(str(e))` or so below)

**Comment by The-Compiler** [comment]
Please also add a test to `tests/end2end/features/misc.feature` - I'd say one with a file which exists and one which doesn't. You can use `(testdata)` to refer to the test data directory.

File: qutebrowser/browser/commands.py [hunk 2]
```diff
@@ -1927,12 +1927,12 @@ def debug_webaction(self, action, count=1):
...
+    def jseval(self, js_code, file=False, quiet=False, *,
                world: typing.Union[usertypes.JsWorld, int]=None):
         """Evaluate a JavaScript string.
 
         Args:
-            js_code: The string to evaluate.
+            js_code: The string/file to evaluate.
             quiet: Don't show resulting JS object.
             world: Ignored on QtWebKit. On QtWebEngine, a world ID or name to
                    run the snippet in.
```
Comment (The-Compiler) [inline qutebrowser/browser/commands.py]: You'll also need to describe `file` here (and add tests like mentioned above :wink:)

**Comment by pkill-nine** [comment]
asciidoc2html.py doesn't seem to be registering this flag

File: qutebrowser/browser/commands.py [hunk 3]
```diff
@@ -1960,6 +1961,14 @@ def jseval_cb(out):
                         out = out[:5000] + ' [...trimmed...]'
                     message.info(out)
 
+        if file:
+            try:
+                with open(js_code, 'r', encoding='utf-8') as f:
+                    js_code = f.read()
+            except OSError as e:
+                message.error(str(e))
+                return
```
Comment (The-Compiler) [inline qutebrowser/browser/commands.py]: I didn't see this earlier, but you can actually do `raise cmdexc.CommandError(str(e))` here.

**Pull request merged**